sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateSchemaCompatibilityChecker.scala

```diff
@@ -71,20 +72,16 @@ class StateSchemaCompatibilityChecker(
   private def schemasCompatible(storedSchema: StructType, schema: StructType): Boolean =
     DataType.equalsIgnoreNameAndCompatibleNullability(storedSchema, schema)
 
-  private def readSchemaFile(): (StructType, StructType) = {
+  // Visible for testing
+  private[sql] def readSchemaFile(): (StructType, StructType) = {
     val inStream = fm.open(schemaFileLocation)
     try {
       val versionStr = inStream.readUTF()
-      // Currently we only support version 1, which we can simplify the version validation and
-      // the parse logic.
       val version = MetadataVersionUtil.validateVersion(versionStr,
         StateSchemaCompatibilityChecker.VERSION)
-      require(version == 1)
-
-      val keySchemaStr = inStream.readUTF()
-      val valueSchemaStr = inStream.readUTF()
-
-      (StructType.fromString(keySchemaStr), StructType.fromString(valueSchemaStr))
+      require(1 <= version && version <= StateSchemaCompatibilityChecker.VERSION)
```

Comment:
`MetadataVersionUtil.validateVersion` does the thing actually. We should also have the check in reader factory as well so kind of redundant.

Reply:
I'll remove `require` here and add a validation for reader.

Reply:
I remove `require` here and how about moving `MetadataVersionUtil.validateVersion` from here to `SchemaReaderFactory.createSchemaReader`?

Reply:
MetadataVersionUtil.validateVersion is also used in HDFSMetadataLog, hence it landed to the utility class.

Reply:
I've moved the invocation of `MetadataVersionUtil.validateVersion` to `SchemaReader.createSchemaReader` and `SchemaWriter.createSchemaWriter`.